modules/admin-ui-picard/app/src/components/events/partials/EditMetadataEventsModal.js

```diff
@@ -0,0 +1,263 @@
+import cn from "classnames";
+import RenderField from "./wizards/RenderField";
+import {postEditMetadata, updateBulkMetadata} from "../../../thunks/eventThunks";
+import RenderMultiField from "../../shared/wizard/RenderMultiField";
+
+
+// Get info about the current language and its date locale
+const currentLanguage = getCurrentLanguageInformation();
+
+/**
+ * This component manges the edit metadata bulk action
+ */
+const EditMetadataEventsModal = ({ close, selectedRows, updateBulkMetadata}) => {
+    const { t } = useTranslation();
+
+    const [selectedEvents, setSelectedEvents] = useState(selectedRows);
+    const [metadataFields, setMetadataFields] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [fatalError, setFatalError] = useState({});
+    const [fetchedValues, setFetchedValues] = useState(null);
+
+    useEffect(() => {
+        async function fetchData () {
+            setLoading(true);
+
+            let eventIds = [];
+            selectedEvents.forEach(event => eventIds.push(event.id));
+
+            // Get merged metadata from backend
+            const responseMetadataFields = await postEditMetadata(eventIds);
+
+            // Set fatal error if response contain error
+            if (!!responseMetadataFields.fatalError) {
+                setFatalError(responseMetadataFields);
+            } else {
+                // Set initial values and save metadata field infos in state
+                let initialValues = getInitialValues(responseMetadataFields);
+                setFetchedValues(initialValues);
+                setMetadataFields(responseMetadataFields);
+            }
+            setLoading(false);
+        }
+        fetchData();
+    }, []);
+
+    const handleSubmit = values => {
+        const response = updateBulkMetadata(metadataFields, values);
+        close();
+    };
+
+    const onChangeSelected = (e, fieldId) => {
+        let selected = e.target.checked;
+        let fields = metadataFields;
+        fields.mergedMetadata = metadataFields.mergedMetadata.map(field => {
+            if (field.id === fieldId) {
+                return {
+                    ...field,
+                    selected: selected
+                };
+            } else {
+                return field;
+            }
+        });
+
+        setMetadataFields(fields);
+    };
+
+    // Check if value of metadata field is changed
+    const isTouchedOrSelected = (field, formikValues) => {
+        if (field.selected) {
+            return true;
+        }
+
+        if (fetchedValues[field.id] !== formikValues[field.id]) {
+            let fields = metadataFields;
+            fields.mergedMetadata = metadataFields.mergedMetadata.map(f => {
+                if (f.id === field.id) {
+                    return {
+                        ...f,
+                        selected: true
+                    };
+                } else {
+                    return f;
+                }
+            });
```

Comment:
This section of the code currently leads to edited fields being impossible to deselect. I'm pretty sure, that isn't the desired functionality

Reply:
Yes, for the moment this is the desired functionality. EditMetadata-Modal needs a revisit when the edit multi-field problem is solved